tools/accuracy_checker/accuracy_checker/postprocessor/nms.py

```diff
@@ -230,3 +230,93 @@ def _nms(self, input_bboxes, input_scores):
             scores *= np.exp(np.negative(np.square(iou_values) / self.sigma))
 
         return np.array(out_ids, dtype=np.int32), np.array(out_scores, dtype=np.float32)
+
+class DIOU_NMS(Postprocessor):
+    __provider__ = 'diou_nms'
+
+    prediction_types = (DetectionPrediction, ActionDetectionPrediction)
+    annotation_types = (DetectionAnnotation, ActionDetectionPrediction)
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'overlap': NumberField(
+                min_value=0, max_value=1, optional=True, default=0.5,
+                description="Overlap threshold for merging detections."
+            ),
+            'include_boundaries': BoolField(
+                optional=True, default=True, description="Shows if boundaries are included."
+            ),
+            'keep_top_k': NumberField(min_value=0, optional=True, description="Keep top K."),
+            'use_min_area': BoolField(
+                optional=True, default=False,
+                description="Use minimum area of two bounding boxes as base area to calculate overlap"
+            )
+        })
+        return parameters
+
+    def configure(self):
+        self.overlap = self.get_value_from_config('overlap')
+        self.include_boundaries = self.get_value_from_config('include_boundaries')
+        self.keep_top_k = self.get_value_from_config('keep_top_k')
+        self.use_min_area = self.get_value_from_config('use_min_area')
+
+    def process_image(self, annotations, predictions):
+        for prediction in predictions:
+            scores = get_scores(prediction)
+            keep = self.diou_nms(
+                prediction.x_mins, prediction.y_mins, prediction.x_maxs, prediction.y_maxs, scores,
+                self.overlap, self.include_boundaries, self.keep_top_k, self.use_min_area
+            )
+            prediction.remove([box for box in range(len(prediction.x_mins)) if box not in keep])
+
+        return annotations, predictions
+
+    @staticmethod
+    def diou_nms(x1, y1, x2, y2, scores, thresh, include_boundaries=True, keep_top_k=None, use_min_area=False):
+        """
+        Pure Python NMS baseline.
+        """
```

Comment:
please remove comment, which is not related to your method

Reply:
done